packages/mobile/src/screens/chat-screen/ChatMessageTrack.tsx

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useMemo } from 'react'
+
+import type { ChatMessageTileProps, ID, TrackPlayback } from '@audius/common'
+import {
+  Kind,
+  PlaybackSource,
+  QueueSource,
+  accountSelectors,
+  getPathFromTrackUrl,
+  makeUid,
+  useGetTrackByPermalink,
+  useToggleTrack
+} from '@audius/common'
+import { useSelector } from 'react-redux'
+
+import { TrackTile } from 'app/components/lineup-tile'
+import { make, track as trackEvent } from 'app/services/analytics'
+
+const { getUserId } = accountSelectors
+
+export const ChatMessageTrack = ({
+  link,
+  onEmpty,
+  onSuccess,
+  styles
+}: ChatMessageTileProps) => {
+  const currentUserId = useSelector(getUserId)
+
+  const permalink = getPathFromTrackUrl(link)
+  const { data: track } = useGetTrackByPermalink(
+    {
+      permalink,
+      currentUserId
+    },
+    { disabled: !permalink }
+  )
+  const item = track
```

Comment:
I know you have TODOs, so probably will be fixed, but same question here about redefining the items on each render.